src/prefect/experimental/workers/base.py

```diff
@@ -473,6 +560,36 @@ async def _propose_crashed_state(self, flow_run: FlowRun, message: str) -> None:
                     f"Reported flow run '{flow_run.id}' as crashed: {message}"
                 )
 
+    async def _set_work_pool_template(self, work_pool, job_template):
+        """Updates the `base_job_template` for the worker's workerpool both server side and locally."""
+        await self._client.update_work_pool(
+            work_pool=work_pool, base_job_template=job_template
+        )
+        work_pool.base_job_template = job_template
+
+    def _create_job_template(self) -> dict:
+        """Create a job template from a worker's configuration components."""
+        if self.job_configuration is None:
```

Comment:
made redundant by removing Optional[job_config] and add test to make sure that's the case